### PR TITLE
feat(utils): add format_ts

### LINKED ISSUE #1085: Let nextcord.utils.format_dt also take a timestamp instead of only datetime
### Summary

Let the `format_dt` function from `nextcord.utils` also be able to work with unix timestamps 

### What is the feature request for?

The core library

### The Problem

I have to either manually format them or convert something to a python `datetime` object just for the sake of using the `format_dt` function

### The Ideal Solution

Let `nextcord.utils.format_dt(...)` also take unix timestamps instead of only `datetime` objects, especially when the `format_dt` function just calls `.timestamp()` on the supplied `datetime` object anyways

### The Current Solution

Manually formatting it when not converting it to a `datetime` object just to be able to use this function

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by teaishealthy**
## Summary

Create a new function `format_ts` that can format a unix timestamp, format_dt can use this function too to reduce code duplication.

Closes #1085 

## This is a **Code Change**

- [x] I have tested my changes.
- [x] I have updated the documentation to reflect the changes.
- [x] I have run `task pyright` and fixed the relevant issues.

### CODE REVIEW

File: nextcord/utils.py [hunk 1]
```diff
@@ -1129,9 +1129,54 @@ def format_dt(dt: datetime.datetime, /, style: Optional[TimestampStyle] = None)
     """
     if not isinstance(dt, datetime.datetime):  # pyright: ignore[reportUnnecessaryIsInstance]
         raise InvalidArgument("'dt' must be of type 'datetime.datetime'")
+    return format_ts(int(dt.timestamp()), style=style)
```
Comment (spifory) [inline nextcord/utils.py:1132]: No need to re-int an integer
```suggestion
    return format_ts(dt.timestamp(), style=style)
```

File: nextcord/utils.py [hunk 2]
```diff
@@ -1129,9 +1129,54 @@ def format_dt(dt: datetime.datetime, /, style: Optional[TimestampStyle] = None)
...
+
+
+def format_ts(ts: int, /, style: Optional[TimestampStyle] = None) -> str:
+    """A helper function to format an :class:`int` unix timestamp for presentation within Discord.
```
Comment (spifory) [inline nextcord/utils.py:1136]:
```suggestion
    """A helper function to format a Unix timestamp as an :class:`int` for presentation within Discord.
```

File: nextcord/utils.py [hunk 3]
```diff
@@ -1129,9 +1129,54 @@ def format_dt(dt: datetime.datetime, /, style: Optional[TimestampStyle] = None)
...
+
+    Note that the exact output depends on the user's locale setting in the client. The example output
+    presented is using the ``en-GB`` locale.
+
+    .. versionadded:: 2.6
+
+    Parameters
+    ----------
+    ts: :class:`int`
+        The unix timestamp to format.
```
Comment (spifory) [inline nextcord/utils.py:1166]:
```suggestion
        The Unix timestamp to format.
```

File: nextcord/utils.py [hunk 4]
```diff
@@ -1129,9 +1129,54 @@ def format_dt(dt: datetime.datetime, /, style: Optional[TimestampStyle] = None)
...
+    style: :class:`str`
+        The style to format the datetime with.
```
Comment (spifory) [inline nextcord/utils.py:1168]:
```suggestion
        The style to format the timestamp with.
```

File: nextcord/utils.py [hunk 5]
```diff
@@ -1129,9 +1129,54 @@ def format_dt(dt: datetime.datetime, /, style: Optional[TimestampStyle] = None)
...
+
+    Returns
+    -------
+    :class:`str`
+        The formatted string.
+    """
+    if not isinstance(ts, int):  # pyright: ignore[reportUnnecessaryIsInstance]
```
Comment (spifory) [inline nextcord/utils.py:1175]: While I do understand that `format_dt` initially did this, _is_ this actually needed? `ts` is already typed as an integer so is there really a need to check if it's an integer a second time?

File: nextcord/utils.py [hunk 6]
```diff
@@ -1129,9 +1129,54 @@ def format_dt(dt: datetime.datetime, /, style: Optional[TimestampStyle] = None)
     """
     if not isinstance(dt, datetime.datetime):  # pyright: ignore[reportUnnecessaryIsInstance]
```
Comment (spifory) [inline nextcord/utils.py:1130]: Same question about checking the type again

**Review by spifory — CHANGES_REQUESTED** [review]
And also it's not too important but you may want to add a `versionchanged` to `format_dt`.

File: nextcord/utils.py (hunk 1 above)
Comment (teaishealthy) [reply nextcord/utils.py:1132]: Its a float

**Comment by teaishealthy** [comment]
> And also it's not too important but you may want to add a `versionchanged` to `format_dt`.
@spifory 
The external api for format_dt did not change, no reason to.

**Review by EmreTech — APPROVED** [review]

**Review by spifory — APPROVED** [review]

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**